hulks/base.py

```diff
@@ -26,8 +28,16 @@ def handle(self, args=None):
         return int(not retval)
 
     def lines_iterator(self, filename):
-        with open(filename) as fp:
-            for line_number, line in enumerate(fp.readlines(), 1):
-                # heuristic, so we dont need to handle all "comment" syntax accross languages
-                if ' noqa' not in line.lower():
-                    yield line_number, line
+        if not mimetypes.guess_type(filename)[0].startswith('text/'):
+                print(f'[WARNING] Trying to iterate lines of non-text file {filename!r}. '
+                      'Double-check commit-hook configuration!', file=sys.stderr)
+        try:
+            with open(filename) as fp:
+                for line_number, line in enumerate(fp.readlines(), 1):
+                    # heuristic, so we dont need to handle all "comment" syntax accross languages
+                    if ' noqa' not in line.lower():
+                        yield line_number, line
+        except UnicodeDecodeError as error:
+            *args, reason = error.args
+            reason += f' at file {filename!r}!'
```

Comment:
f-strings are python3.6 only, I believe it's nicer to keep this lib useful in a wider range of versions.
```suggestion
            reason += ' at file {}!'.format(filename)
```